dev/create-release/release-build.sh

```diff
@@ -130,6 +130,10 @@ else
   fi
 fi
 
+LSOF=lsof
+if ! hash $LSOF 2>/dev/null; then
```

Comment:
I don't know much about this but is command better? https://stackoverflow.com/questions/592620/check-if-a-program-exists-from-a-bash-script not sure what the scripts here usually do (not at my keyboard)

Reply:
I am not sure .. but I found `hash` is used in few places more than `command` within Spark:

```bash
grep -r "hash" . | grep ">/dev/null"
```

```
./dev/run-pip-tests:if hash virtualenv 2>/dev/null && [ ! -n "$USE_CONDA" ]; then
./dev/run-pip-tests:  if hash python2 2>/dev/null; then
./dev/run-pip-tests:  elif hash python 2>/dev/null; then
./dev/run-pip-tests:  if hash python3 2>/dev/null; then
./dev/run-pip-tests:elif hash conda 2>/dev/null; then
./dev/run-pip-tests:if ! hash pip 2>/dev/null; then
./sql/create-docs.sh:if ! hash python 2>/dev/null; then
./sql/create-docs.sh:if ! hash mkdocs 2>/dev/null; then
```

Looks:

```bash
grep -r "command" . | grep ">/dev/null"
```

not used in Spark but .. I am not sure ..

Reply:
I personally use `hash` more somehow codes without a specific reason. If anyone prefers `command` specifically, I can change.

Reply:
I’m used to the hash check now as well, but that’s probably from editing Spark shell scripts more than any particular good practice.

Reply:
If `lsof`'s full path can be different in different OSes as @kiszk's comment https://github.com/apache/spark/pull/19695#issuecomment-342865001, seems `command` can help return the full path of `lsof`?